cdap-ui/app/features/adapters/controllers/tabs/log-ctrl.js

```diff
@@ -23,11 +23,10 @@ angular.module(PKG.name + '.feature.adapters')
     angular.copy(AdapterDetail.params, runsParams);
     angular.copy(AdapterDetail.logsParams, logsParams);
     logsParams.scope = $scope;
-
     AdapterDetail.logsApi.pollLatestRun(logsParams)
       .$promise
       .then(function (runs) {
-        if (runs.length === 0) { return; }
+        if (runs.length === 0 || (runs.length && logsParams.runId === runs[0].runid)) { return; }
```

Comment:
@anwar6953 yes. We need to go for all other features. Will open a separate PR for it in sometime.